engine/src/main/java/com/datatorrent/stram/util/LoggerUtil.java

```diff
@@ -464,4 +474,55 @@ public static boolean removeAppender(String name)
     }
     return names;
   }
+
+  /**
+   * Makes MDC properties
+   */
+  public static void setupMDC()
+  {
+    String node = StramClientUtils.getHostName();
+    MDC.put("node", node == null ? "unknown" : node);
+
+    String value = System.getenv("HADOOP_USER_NAME");
+    if (value != null) {
+      MDC.put("user", value);
+    }
+
+    value = System.getenv("CONTAINER_ID");
+    if (value != null) {
+      MDC.put("containerId", value);
+    }
+
+    value = System.getProperties().getProperty(APPLICATION_NAME.getLongName());
+    if (value != null) {
+      MDC.put("application", value);
+    }
+
+    if (MDC.get("service") == null) {
+      MDC.put("service", getService());
+    }
+  }
+
+  /**
+   * Returns the service name
+   * @return Service name
+   */
+  private static String getService()
```

Comment:
I'm sure the implicit setting of the MDC properties with the default values is much convenient and the right approach. If some caller needs to setup the MDC properties explicitly, the caller can call the methods setupMDC() and setServiceName()  explicitly.

And I don't think "the implicit setup makes it much harder to track" is the right statement. It is absolutely generic approach in the Java development to set the default properties values.

I think 3 locations (client, master, worker) are "multiple locations".